### ISSUE TITLE
create a basic UI which accesses the API exposed and returns the rant

### ISSUE THREAD

**Issue by prodicus** [prodicus/trumporate#1]
An Initial wireframe I slapped up on the whiteboard for a possible UI/UX for trumporate

![img_20170413_173837503](https://cloud.githubusercontent.com/assets/4672518/25556020/4e289b16-2d12-11e7-9a5f-2da60ca6c40c.jpg)

**Comment by riturajDas** [comment]
## UI Iteration No. 1
![image](https://cloud.githubusercontent.com/assets/20974909/26600536/1444e094-459a-11e7-8338-bba26a4c9317.png)

**Comment by prodicus** [comment]
Looks good!

Some comments
- Is the Share button functional right now? If yes, what happens when I press Share? Do media buttons pop up?
- the theme looks pretty serious to me. What do you think?

**Comment by riturajDas** [comment]
Alrighty then!
- The buttons need some JS to be manipulating the window atm. But, if you put your pointer onto the buttons, they will fade-in/out for 2sec though CSS animation only.
- What does the media button do? Do we need a media button? wouldn't seem cool anymore if cluttered by buttons/pop-ups.
- Neither of us is a designer, so you'll have to do a better job at explaining what kinda theme you want. noh? I looked into several Codepen samples you sent earlier, this is my "inspired by" version.
- I will put a "Made with ♥️ in India by *" footer soon after 👆🏽.

**Comment by prodicus** [comment]
> What does the media button do? Do we need a media button? wouldn't seem cool anymore if cluttered by buttons/pop-ups.

Just a twitter share button should do I feel.

> Neither of us is a designer, so you'll have to do a better job at explaining what kinda theme you want. noh ? I looked into several Codepen samples you sent earlier, this is my "inspired by" version.

😂 That's ok man. Did share some possible layouts. The clean ones. The which we have right now is bad per se but I feel we can iterate on it. Whatsay? Like the blog URL which I shared. 

> I will put a "Made with ♥️ in India by *" footer soon after 👆🏽.

Great! Please post the newer screenshot after the modifications you have made here. 

> How about we link our respective portfolios to our names in the footer text ?

Yep. That should not be problem.

**Comment by riturajDas** [comment]
- "Twitter share button" ok, you got it 👍🏼.

> 😂 That's ok man. Did share some possible layouts. The clean ones. The which we have right now is bad per se but I feel we can iterate on it. Whatsay? Like the blog URL which I shared.

- Define "bad". But we can definitely iterate, right? share a different vibe?

**Comment by prodicus** [comment]
@riturajDas 

Daym Typo. I correct. *not bad per se. 

My bad my bad.

**Comment by riturajDas** [comment]
😂 ginormous typo bruh.

**Comment by prodicus** [comment]
Sorry man. My bad 😂

**Comment by riturajDas** [comment]
## UI Iteration No. 2
![screen shot 2017-06-01 at 3 26 48 am](https://cloud.githubusercontent.com/assets/20974909/26655998/da4c47f6-467a-11e7-80c9-f2e88eab935a.jpg)
- Background transitions into random color in 5sec intervals automagically.
- "Refresh" icon spins infinitely unlike a fidget spinner.
- Before icons load up, they say "firing up engines...".
### What JS does atm
- Generates random RGBA-space colors.
### What JS doesn't atm
- Fetch next quote/rant.
- Actually share/tweet content onto user-space

**Comment by riturajDas** [comment]
## Branch Issue_1
![screen shot 2017-06-01 at 4 43 16 am](https://cloud.githubusercontent.com/assets/20974909/26657998/f9e9a946-4684-11e7-8829-ab371ee0aac4.jpg)

Is this what you were saying earlier when you said "refactoring thingy". Will make PR after working AJAX. Whatsay?

**Comment by prodicus** [comment]
@riturajDas 

About the refactoring part, the folders `templates`, `static`should be place the package `trumporate`

> Is this what you were saying earlier when you said "refactoring thingy". Will make PR after working AJAX. Whatsay?

that would be fine.

> UI Iteration No. 2

Looks good. 

My input 

- Background, this one should do. 
- As for the font, maybe we can try out different ones?
- from what I have seen, most of the websites place the buttons aligned to the right. My observation was that.

Good work!

**Comment by riturajDas** [comment]
Oh, the one one inside u mean? ok.

**Comment by prodicus** [comment]
> Oh, the "trumporate" inside "trumporate" you mean? but isn't this correct?

Somewhat like this, taking [plino](https://github.com/prodicus/plino) as reference here

```bash
tasrahma at dev-box in ~/development/projects/plino (master)
$ tree -L 3
├── plino
│   ├── __init__.py
│   ├── app.py
│   ├── classifier.py
│   ├── nltk_data
│   │   └── corpora
│   ├── saved_classifiers
│   │   ├── __init__.py
│   │   ├── spam_classifier.pickle
│   │   └── trainer.pickle
│   ├── static
│   │   ├── css
│   │   ├── favicon.ico
│   │   ├── fonts
│   │   ├── images
│   │   ├── img
│   │   ├── js
│   │   └── less
│   ├── templates
│   │   ├── 404.html
│   │   └── index.html
│   ├── train.py
│   └── utils.py
├── requirements.txt
└── tests
    └── test_plino_app_api_response.py
```

> I agree that the buttons should do more work if placed on the right-end of the content but I'm afraid that might somewhat look inconsistent to the "all left-aligned" approach.

As you wish, your call.

> We could add more buttons though, show few more options/love for Trumporate? 😋

Twitter button should do IMO. Although, if you have other buttons in mind, I am all ears

**Comment by riturajDas** [comment]
Look at this 

```
Riturajs-MacBook-Pro:trumporate riturajdas$ tree -L 3
.
├── LICENSE
├── Makefile
├── README.md
├── ansible
│   ├── README.md
│   ├── ansible.cfg
│   ├── inventory.example
│   ├── roles
│   │   ├── install_packages
│   │   ├── nginx
│   │   ├── provision
│   │   └── systemd
│   └── site.yml
├── corpus
│   ├── facebook
│   │   ├── Donald_Trumps_Facebook_Statuses_+_Reaction_Counts_(as_of_10%2F17%2F16_@minimaxir)_DonaldTrump_facebook_statuses.csv.csv
│   │   ├── donald_duck_statuses.txt
│   │   └── source.txt
│   ├── presidential16_speeches
│   │   ├── source.txt
│   │   └── speeches.txt
│   ├── rants_2000.txt
│   └── twitter
│       ├── 2016_12_05-TrumpTwitterAll.xlsx
│       ├── 2017_01_28_Trump_Tweets.xlsx
│       ├── source.txt
│       └── trumptweets1205-127.csv
├── requirements-dev.txt
├── scripts
│   ├── __init__.py
│   └── generate_rants.py
├── static
│   ├── css
│   │   └── style.css
│   ├── img
│   │   ├── Blackboard_Chalkboard_Wipemarks_UV_CM_1.jpg
│   │   ├── Donald\ Trump_100.png
│   │   ├── Donald\ Trump_50.png
│   │   ├── cropped-1920-1080-708195.jpeg
│   │   ├── cropped-1920-1080-781782.jpg
│   │   ├── donald-trump-cpac-800x430.jpg
│   │   └── name.png
│   └── js
│       └── app.js
├── templates
│   └── index.html
└── trumporate
    ├── __init__.py
    ├── api
    │   ├── __init__.py
    │   └── v1
    └── wsgi.py

19 directories, 33 files
```
this how "issue_1" looks on mine. Please advise.

**Comment by prodicus** [comment]
The preferable structuring would be 

```bash
└── trumporate
    ├── __init__.py
    ├── templates/*
    ├── static/*
    ├── api
    │   ├── __init__.py
    │   └── v1
    └── wsgi.py
```

**Comment by riturajDas** [comment]
ohh, ok 👍🏼 wilco.

**Comment by riturajDas** [comment]
Package structure refactored now 👍🏼

**Comment by riturajDas** [comment]
## Improved font scaling and visibility in lighter bg introduced
![screen shot 2017-06-01 at 7 48 42 pm](https://cloud.githubusercontent.com/assets/20974909/26684028/762009f0-4703-11e7-824a-df59a10e3ff4.jpg)

**Comment by prodicus** [comment]
The heading (can be placed on the wireframe position)"Trumporate" should be
bigger than the text. What do you think? Making it bold makes the footer
look invisible imo

Regards,
Tasdik

[image: http://]*tasdikrahman.me*
<http://tasdikrahman.me/>

[image: https://in.linkedin.com/in/tasdikrahman]
<https://in.linkedin.com/in/tasdikrahman>   <http://github.com/prodicus>
<http://www.quora.com/Tasdik-Rahman>
<https://plus.google.com/+TasdikRahman>   <https://twitter.com/tasdikrahman>

**Comment by riturajDas** [comment]
I tried that but then with this theme, the "Trumporate" header steals the content's attention.
What are those links for btw? not really getting the context of all those links 🤔
## As suggested
![screen shot 2017-06-01 at 8 02 58 pm](https://cloud.githubusercontent.com/assets/20974909/26684753/988e9c2a-4705-11e7-8941-f2a24450760b.jpg)
Cooler w/larger header or just bold in the earlier smaller size-scaling way?

**Comment by prodicus** [comment]
> Cooler w/larger header or just bold in the earlier smaller size-scaling way?

This looks good, the color goes with the buttons too.

> What are those links for btw? not really getting the context of all those links 🤔

Which links? the footer?

**Comment by riturajDas** [comment]
> This looks good, the color goes with the buttons too.
- What color? it's still doing random spectrum cycling for the background.
> Which links? the footer?
- NO, the ones that came in with your earlier comment like links to your blog, linkedin and stuff.

**Comment by prodicus** [comment]
> What color? it's still doing random spectrum cycling for the background.

I meant the trumporate header. 

> links to your blog, linkedin and stuff.

When was that? 🤔 And I think that would be unnecessary. Just having the names pointing to out websites would be enough

**Comment by riturajDas** [comment]
> When was that? 🤔 And I think that would be unnecessary. Just having the names pointing to out websites would be enough
- Don't know where all those links came from (maybe you texted via your gmail client to github). idk there were a lot of your links that confused me as to what you were meaning by "The heading (can be placed on the wireframe position)"Trumporate" should be
bigger than the text. What do you think? Making it bold makes the footer
look invisible imo" w/your links 🤔

**Comment by prodicus** [comment]
> Don't know where all those links came from (maybe you texted via your gmail client to github). 

Replying through email must have been the case. Forget that. 

> "The heading (can be placed on the wireframe position)"Trumporate" should be
bigger than the text. What do you think? Making it bold makes the footer
look invisible imo" w/your links 🤔

The last iteration looks good. The only thing that I feel is that the alignment of the thing should be in the middle. As in center. Gives a better geometry imo. What do you think? 

The fonts on the current iteration, also we can work on changing and trying out different font's(fonts as in san-serif not the text size if you are confusing with that). The blog I showed had some good fonts for inspiration

**Comment by riturajDas** [comment]
- Please decide on one bruh : Left( like Trey Huffine's portfolio) or symmetric/centered.
- Current font is a serif derivative. The one used in Trey Huffine's portfolio is Raleway( material design favourite among Roboto) I think. Take a look at Google fonts once, tell me if you find one you like.

**Comment by riturajDas** [comment]
## Update: quote icons
![screen shot 2017-06-02 at 8 41 56 pm](https://cloud.githubusercontent.com/assets/20974909/26732116/18d10ca0-47d4-11e7-97d5-4bf4cd091999.jpg)

**Comment by prodicus** [comment]
This background looks amaze 💯

**Comment by prodicus** [comment]
> Please decide on one bruh : Left( like Trey Huffine's portfolio) or symmetric/centered.

I would say, post screens of both here and we can have a better comparison then.

> Current font is a serif derivative. The one used in Trey Huffine's portfolio is Raleway( material design favourite among Roboto) I think. Take a look at Google fonts once, tell me if you find one you like.

Some fonts which drew my attention from [here](https://fonts.google.com/) at first whim, do take a look at the others too

- Bubbler One
- Lora
- Abel
- Pacifico (Try the trumporate header in this, might look good)
- Signika
- Dancing script
- Domine

Others would be great too, it's already sorted a/c to most popular

**Comment by riturajDas** [comment]
Lol, I thought you didn't like goofy handwriting-like fonts. All right, will keep you updated, was busy with Nokia's 2nd round test earlier today. Will resume now 👍🏼

**Comment by prodicus** [comment]
Ah, no prob man. Whenever you get free.

**Comment by riturajDas** [comment]
- Loooks ok imo
![screen shot 2017-06-02 at 9 14 28 pm](https://cloud.githubusercontent.com/assets/20974909/26733463/84921188-47d8-11e7-9e79-4c9198520222.jpg)

- The github corner is creating problems for things at the top.
![screen shot 2017-06-02 at 9 18 06 pm](https://cloud.githubusercontent.com/assets/20974909/26733631/09df943c-47d9-11e7-92fa-5eb91a991bbe.jpg)

**Comment by prodicus** [comment]
The 2nd image background again looks good. I think we can try going for a consistent color. The trumporate header would need some work imo. Different style maybe? The button at the bottom can be tried to be in alignment of the right most border for the text

**Comment by prodicus** [comment]
As for the gh-button I think we can put it down on the footer somewhere on the right. Might look good. Or leave it for now. We can have it placed later too just before releasing

**Comment by riturajDas** [comment]
## Font updates
![screen shot 2017-06-02 at 11 11 52 pm](https://cloud.githubusercontent.com/assets/20974909/26737721/f7f417a6-47e8-11e7-8701-13c0ae834e2c.jpg)

## Also added a little Trump icon for the tab.
![screen shot 2017-06-02 at 11 15 58 pm](https://cloud.githubusercontent.com/assets/20974909/26737861/94ae7b5e-47e9-11e7-9f15-5c42b679e74c.jpg)

Looks crazy delicious to mee 🤑

**Comment by prodicus** [comment]
Good work 💯 
I would say, we are good to go 👍 
Serious thought, this last blue color is going really well, we should fix the bg color to this

**Comment by prodicus** [comment]
Also, we can put powered by  [markovipy](https://github.com/prodicus/markovipy) somewhere at the bottom

**Comment by riturajDas** [comment]
> Serious thought, this last blue color is going really well, we should fix the bg color to this
- Could you point it out?

> Also, we can put powered by markovipy somewhere at the bottom
- sure 👍

**Comment by prodicus** [comment]
Looks like you the image of the particular image. 

https://cloud.githubusercontent.com/assets/20974909/26737861/94ae7b5e-47e9-11e7-9f15-5c42b679e74c.jpg

**Comment by riturajDas** [comment]
owh, lol. I kinda removed that one. Ok. No prob (y)

**Comment by prodicus** [comment]
(y)

**Comment by riturajDas** [comment]
## Successfully fetching rants via crossorigin.me
![screen shot 2017-06-16 at 2 35 53 am](https://user-images.githubusercontent.com/20974909/27201822-c0d82f88-523c-11e7-88a7-d22ca7af4812.jpg)
### what is left 
- refresing rants using the spinning button,
- share rants on twitter.

**Comment by riturajDas** [comment]
## It works 😃👍
![screen shot 2017-06-16 at 3 53 58 am](https://user-images.githubusercontent.com/20974909/27204242-9e1e9fd0-5247-11e7-97fa-5b034ef2170f.jpg)
# So what works? these puppies bork hard :
- Spinning refresh rant button works 💯
- Twitter-share button says "Share awesome Trump rants using Trumporate. Check it out! at http://trumporate.com/whatevs" as filler content. Will fix the filler later.

**Comment by prodicus** [comment]
@riturajDas Good work! 💯 

Two things I was curious about
- the rant text, does it get presented all at once or is there an animation from left to write as if someone is writing it out(we discussed something like that sometime back)
- the newer tweet button feels a little bit off as the new rant button is white in color. What happened to the one in the last screenshot? 

Also, 

I would suggest now, make a PR to the master if all the things are done.

**Comment by riturajDas** [comment]
> the rant text, does it get presented all at once or is there an animation from left to write as if someone is writing it out(we discussed something like that sometime back)
-Yeah, will put some animation as to how the text rolls out on screen.

> the newer tweet button feels a little bit off as the new rant button is white in color. What happened to the one in the last screenshot?
- Lol, I actually think it looks ugly on our theme but this is Twitter's own implementation. Will find a workaround this hideous button and load custom styles.

>I would suggest now, make a PR to the master if all the things are done.
-Sure 👍🏽

**Comment by prodicus** [comment]
> Yeah, will put some animation as to how the text rolls out on screen.

Was just suggesting so only if it looks good, otherwise drop the animation thing.

> Will find a workaround this hideous button and load custom styles.

The old button looks good imo, we can put that.

**Comment by riturajDas** [comment]
## Twitter button restored to the 
![screen shot 2017-06-16 at 5 18 37 pm](https://user-images.githubusercontent.com/20974909/27225479-ff66a844-52b7-11e7-898d-39548567844d.jpg)
previous state
> I gotcha homie 😉

**Comment by riturajDas** [comment]
## To-do
- [ ] Link to Trumporate on the github corner.

**Comment by prodicus** [comment]
Added that in the latest commit and updated README.

**Comment by prodicus** [comment]
closed by #9
